include/envoy/server/filter_config.h

```diff
@@ -200,6 +200,20 @@ class FactoryContext : public virtual CommonFactoryContext {
   virtual ProtobufMessage::ValidationVisitor& messageValidationVisitor() PURE;
 };
 
+/**
+ * An implementation of FactoryContext. It should be used to create NetworkFilterChain.
+ */
+class FilterChainFactoryContext : public virtual FactoryContext {
+public:
+  /**
+   * @return uint64_t The tag assigned with this context at runtime.
```

Comment:
Ditto, it's unclear what tags are for.

Reply:
An example of tag usage:
* The initial listener has filter chain A, A will be assigned the tag value 1. The workers see the tag set {1}
* The second listener config update with filter chain A and B.  
1. Best: A assigned tag value 1, B assigned with tag value 2. The worker will not drain the connection belongs to A because the new tag set is {1,2} 
2. Bad: A assigned tag value 2, B assigned with tag value 3. The worker will drain the connection belongs to old A because the new tag set is {2,3} and tag 1 is not in.